File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/localization/OdometrySet.java
package org.firstinspires.ftc.teamcode.control.localization;

import org.openftc.revextensions2.ExpansionHubMotor;

public class OdometrySet {
    private ExpansionHubMotor vertical, horizontal;
    private int verticalOffset = 0;
    private int horizontalOffset = 0;

    public OdometrySet(ExpansionHubMotor vertical, ExpansionHubMotor horizontal) {
        this.vertical = vertical;
        this.horizontal = horizontal;
        markCurrOffset();
    }

    public void markCurrOffset() {
        verticalOffset = vertical.getCurrentPosition();
        horizontalOffset = horizontal.getCurrentPosition();
    }

    public int getVerticalTicks() {
        return vertical.getCurrentPosition() - verticalOffset;
    }

    public int getHorizontalTicks() {
        return horizontal.getCurrentPosition() - horizontalOffset;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/localization/OdometrySet.kt
package org.firstinspires.ftc.teamcode.control.localization

import org.openftc.revextensions2.ExpansionHubMotor

data class OdometrySet(
    private val vertical: ExpansionHubMotor,
    private val horizontal: ExpansionHubMotor
) {
    private var verticalOffset = 0
    private var horizontalOffset = 0
    private fun markCurrOffset() {
        verticalOffset = vertical.currentPosition
        horizontalOffset = horizontal.currentPosition
    }

    val verticalTicks: Int
        get() = vertical.currentPosition - verticalOffset
    val horizontalTicks: Int
        get() = horizontal.currentPosition - horizontalOffset

    init {
        markCurrOffset()
    }
}
